apache/superset
# fix: Trino - handle table not found in SQLLab

### SUMMARY


SQLAlchemy doesn't strictly define what `get_indexes` method should do when table is not found.
Some dialects return the empty list while some of them raise `NoSuchTableError`.

When `NoSuchTableError` error is thrown, API returns 500 when non-existing table details are requested.
This PR fixes this issue and returns 404 in such cases.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


### TESTING INSTRUCTIONS


Paste some non-existing table name in table name field and hit Enter:

<img width="1253" alt="Screenshot 2023-12-25 at 18 33 10" src="https://github.com/apache/superset/assets/3982146/a9a3b9aa-7761-4ca0-8ed0-ad009ca80968">


Expected: no 500 errors from the backend.

### ADDITIONAL INFORMATION
